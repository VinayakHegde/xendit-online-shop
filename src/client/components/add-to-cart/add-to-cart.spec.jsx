import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import * as ReactRedux from 'react-redux';
import { AddToCart } from ".";


const spyOnUseDispatch = (mock, expected) =>
  jest.spyOn(ReactRedux, 'useDispatch').mockImplementation(() => mock.mockResolvedValue(expected));

afterAll(() => {
  jest.clearAllMocks();
});
const product = (quantity) => ({
  quantity
});
describe.each([
  ['enabled', 10],
  ['disabled', 0]
])('AddToCart - %s', (a, b) => {

  it('should render without crash', () => {
    const mockDispatch = jest.fn();
    spyOnUseDispatch(mockDispatch, () => { });
    const { container } = render(<AddToCart product={product(b)} classPrefix="ProductDetails" />)
    expect(container).toMatchSnapshot();
  });
  it('should fire click event', () => {
    const mockDispatch = jest.fn();
    spyOnUseDispatch(mockDispatch, () => { });
    const { getByTitle } = render(<AddToCart product={product(b)} classPrefix="ProductDetails" />)
    const button = getByTitle('Add to cart');
    act(() => {
      fireEvent.click(button);
    });
    if (b) {
      expect(mockDispatch).toHaveBeenCalled();
    } else {
      expect(mockDispatch).not.toHaveBeenCalled();
    }
  });
});
