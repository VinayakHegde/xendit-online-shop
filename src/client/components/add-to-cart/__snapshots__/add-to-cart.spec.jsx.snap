// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCart - disabled should render without crash 1`] = `
<div>
  <button
    class="Button AddToCart ProductDetails__button"
    disabled=""
    title="Add to cart"
    type="button"
  >
    Add to cart
  </button>
</div>
`;

exports[`AddToCart - enabled should render without crash 1`] = `
<div>
  <button
    class="Button AddToCart ProductDetails__button"
    title="Add to cart"
    type="button"
  >
    Add to cart
  </button>
</div>
`;
